### ISSUE TITLE
Quick Open is too slow

### ISSUE THREAD

**Issue by paolo-venturi** [microsoft/vscode#76448]
VSCode Version: 1.35 both regular and insider
OS Version: Win 8/10

Steps to Reproduce:
- Ctrl+P to pull up quick action box
- start typing in filename in the project
- expect near instant response of list of files matches as in VSCode 1.34
- actual behavior is shows the revolving progress bar for 20-30 seconds before showing the list

Our project is about 20000 files.

I tried to disable the parameters indicated in this guide https://github.com/microsoft/vscode/wiki/Search-Issues#search-including-too-many-results (search.useIgnoreFiles, search.followSymlinks) but the problem continues to exist. 
In previous versions Quick Open was very fast.

Does this issue occur when all extensions are disabled?: Yes

**Comment by roblourens** [comment]
Nothing has changed recently. Just to check, you say you disabled `search.useIgnoreFiles`, disabling it can make search much slower. It is enabled by default and you probably want it enabled.

Is this due to windows defender running while search is running? https://github.com/microsoft/vscode/wiki/Search-Issues#slow-search-rg-running-for-a-long-time-or-consuming-lots-of-cpumemory

**Comment by paolo-venturi** [comment]
I tried to enable search.useIgnoreFiles but the search is still slow.

I attach a zipped video whit all the extensions disabled to show how it's slow.

[Video_2019-07-04_085359.zip](https://github.com/microsoft/vscode/files/3358074/Video_2019-07-04_085359.zip)

**Comment by roblourens** [comment]
It kind of looks like something else must be slowing down your machine, just because even the UI of vscode is so slow. Maybe check the task manager.

**Comment by paolo-venturi** [comment]
I check the task manager: Vs Code normally uses 5% of CPU resources, when i start a Quick Open search 
usage increases to 30/50%.
The tests were done with all PC applications closed.

I attach a zipped video of 1.22.2 version on the same PC whit all the extensions disabled to show how fast it was.
[Video_2019-07-10_092037.zip](https://github.com/microsoft/vscode/files/3376220/Video_2019-07-10_092037.zip)

**Comment by roblourens** [comment]
This one simply shows a search finishing immediately. When the "file results" appear, the search is complete and vscode should not be using lots of CPU at that point.

**Comment by paolo-venturi** [comment]
https://github.com/microsoft/vscode/files/3358074/Video_2019-07-04_085359.zip show a search of word "xml" in our project folder with VS Code Insiders 1.37.0. The search ends in about 20 seconds and the CPU usage increases to 30/50%.

https://github.com/microsoft/vscode/files/3376220/Video_2019-07-10_092037.zip show a search of word "xml" in the same project folder with VS Code 1.22.2. The search end immediately and if there is an increase in CPU usage it is too fast to be noticed.

We did some tests and it seems that the problem occurs with the verisons from 1.35.0 onwards.
Versions 1.34.0 and earlier do not show slowness.

**Comment by roblourens** [comment]
Strange, thanks for testing with old versions.

Which processes are using high CPU? You can run `code --status` to get a listing while the search is running. Or check Help > Process Explorer.

**Comment by paolo-venturi** [comment]
Progress bar and blinking cursor sometimes appear sometimes not.

[2019-07-15.zip](https://github.com/microsoft/vscode/files/3391140/2019-07-15.zip)

I recorded Process Explorer of Code Version: 1.36.1, Commit: 2213894ea0415ee8c85c5eea0d0ff81ecc191529 (code.wmv) which has the slow open.

With the last version of Code Insiders (Version: 1.37.0-insider to commit: 894056d11c7fffb261a0100fd4dcbf8cff84a8c0) the problem seems to be solved. I have recorded Process Explorer anyway (insiders.wmv) to compare the results.

**Comment by roblourens** [comment]
Thanks. Since the renderer process (the window) was the one using lots of CPU, and not search, this could have been fixed by an electron update or some other unrelated change. Thanks for trying insiders. I'll consider it closed but let me know if it comes up again.

**Comment by curtisgibby** [comment]
I'm having the same issue on Linux Mint -- very very slow response from Quick Open.

```
Version: 1.36.1
Commit: 2213894ea0415ee8c85c5eea0d0ff81ecc191529
Date: 2019-07-08T22:55:08.091Z
Electron: 4.2.5
Chrome: 69.0.3497.128
Node.js: 10.11.0
V8: 6.9.427.31-electron.0
OS: Linux x64 4.15.0-54-generic
```

From @paolo-venturi 's testing, it appears that this is resolved in 1.37.0. When can we anticipate this will hit general release?

**Comment by paolo-venturi** [comment]
With version 1.37.0 the problem seems to be solved but with the update 1.37.1 the slowdown returns.

With the insider version 1.38.0 everything works.
